### PR TITLE
feat(core): improve question to better support CLI

### PR DESCRIPTION
**Pull request by jayzhang**
1. automatically map question model to CLI options/arguments (in precommit stage of fx-core)
2. Seperate create project into two APIs: `create from scratch` and `create from sample`

Feature link: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24539544

Sub part of this PR: #9499

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -68,14 +68,14 @@
   "plugins.spfx.questions.framework.title": "Framework",
   "plugins.spfx.questions.webpartName.error.duplicate": "The folder %s already exists. Choose a different name for your component.",
   "plugins.spfx.questions.webpartName.error.notMatch": "%s doesn't match pattern: %s",
-  "plugins.spfx.questions.packageSelect.title": "SharePoint Framework",
+  "plugins.spfx.questions.packageSelect.title": "Install latest SharePoint Framework version.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:71]: Install the latest version of SharePoint Framework.

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -291,7 +291,7 @@
   "core.createProjectQuestion.apiSpec.multipleValidationErrors.vscode.message": "Invalid API specification. Check [output panel](command:fx-extension.showOutputChannel) for details.",
   "core.createProjectQuestion.openAiPluginManifest.multipleValidationErrors.message": "Invalid OpenAI plugin manifest. Check output panel for details.",
   "core.createProjectQuestion.openAiPluginManifest.multipleValidationErrors.vscode.message": "Invalid OpenAI plugin manifest. Check [output panel](command:fx-extension.showOutputChannel) for details.",
-  "core.spfxFolder.title": "SPFx solution folder",
+  "core.spfxFolder.title": "Directory path that contains the existing SarePoint Framework solutions.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:294]: Directory path -> Directory or Path, 
solutions -> solution

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -68,14 +68,14 @@
   "plugins.spfx.questions.framework.title": "Framework",
   "plugins.spfx.questions.webpartName.error.duplicate": "The folder %s already exists. Choose a different name for your component.",
   "plugins.spfx.questions.webpartName.error.notMatch": "%s doesn't match pattern: %s",
-  "plugins.spfx.questions.packageSelect.title": "SharePoint Framework",
+  "plugins.spfx.questions.packageSelect.title": "Install the latest version of SharePoint Framework.",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:71]: The question title here is to let user choose which SharePoint Framework he wants to use, either install the latest one locally or use globally installed version. So I don't think the string works
image

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -68,14 +68,14 @@
...
   "plugins.spfx.questions.packageSelect.placeholder": "Select an option for scaffolding",
   "plugins.spfx.questions.packageSelect.useGlobalPackage.withVersion.label": "Use globally installed SPFx (%s)",
   "plugins.spfx.questions.packageSelect.useGlobalPackage.noVersion.label": "Use globally installed SPFx",
   "plugins.spfx.questions.packageSelect.useGlobalPackage.detail": "SPFx %s or later",
   "plugins.spfx.questions.packageSelect.installLocally.withVersion.label": "Install the latest SPFx (%s) locally in Teams Toolkit directory ",
   "plugins.spfx.questions.packageSelect.installLocally.noVersion.label": "Install the latest SPFx locally in Teams Toolkit directory ",
-  "plugins.spfx.questions.spfxSolution.title": "SharePoint Solution",
+  "plugins.spfx.questions.spfxSolution.title": "Create a new or import an existing SharePoint Framework solution.",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:78]: Also add @summzhan here to review the string.

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -291,8 +291,8 @@
   "core.createProjectQuestion.apiSpec.multipleValidationErrors.vscode.message": "Invalid API specification. Check [output panel](command:fx-extension.showOutputChannel) for details.",
   "core.createProjectQuestion.openAiPluginManifest.multipleValidationErrors.message": "Invalid OpenAI plugin manifest. Check output panel for details.",
   "core.createProjectQuestion.openAiPluginManifest.multipleValidationErrors.vscode.message": "Invalid OpenAI plugin manifest. Check [output panel](command:fx-extension.showOutputChannel) for details.",
-  "core.spfxFolder.title": "SPFx solution folder",
-  "core.spfxFolder.placeholder": "Select the folder that contains your SPFx solution",
+  "core.spfxFolder.title": "Directory or Path that contains the existing SharePoint Framework solution.",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:294]: cc @summzhan

File: packages/fx-core/src/question/inputs/CreateSampleProjectInputs.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Inputs } from "@microsoft/teamsfx-api";
+
+export interface CreateSampleProjectInputs extends Inputs {
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/question/inputs/CreateSampleProjectInputs.ts:6]: Will sample owner need to update here if new sample onboarded? cc @tecton

File: packages/fx-core/src/question/options/SFPxAddWebpartOptions.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CLICommandOption, CLICommandArgument } from "@microsoft/teamsfx-api";
+
+export const SFPxAddWebpartOptions: CLICommandOption[] = [
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/question/options/SFPxAddWebpartOptions.ts:6]: Typo: SFPxAddWebpartOptions -> SPFxAddWebpartOptions

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (jayzhang) [reply packages/fx-core/resource/package.nls.json:71]: Yes, you are right, I will use another field to distinguish them.

File: packages/fx-core/src/question/inputs/CreateSampleProjectInputs.ts (hunk 6 above)
Comment (jayzhang) [reply packages/fx-core/src/question/inputs/CreateSampleProjectInputs.ts:6]: This file is automatically generated by our prebuild script. Sample owner don't need to change it manually. But he will be aware of the change once the sample choices are changed. 
I will add some comments on this file.

File: packages/fx-core/src/question/options/SFPxAddWebpartOptions.ts (hunk 7 above)
Comment (jayzhang) [reply packages/fx-core/src/question/options/SFPxAddWebpartOptions.ts:6]: image

File: packages/cli/src/cmds/preview/previewEnv.ts [hunk 8]
```diff
@@ -437,7 +431,7 @@ export default class PreviewEnv extends YargsCommand {
       LogLevel.Warning,
       util.format(constants.manifestChangesHintMessage, `--env ${env}`)
     );
-    if (hub !== Hub.teams) {
+    if (hub !== "teams") {
```
Comment (swatDong) [inline packages/cli/src/cmds/preview/previewEnv.ts:434]: Would it be better to use enum such as `HubTypes.teams`? To prevent typo like `hub !== "team"`

**Review by swatDong — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Review by sffamily — APPROVED** [review]

**Pull request merged**